core/src/main/scala/org/apache/spark/h2o/utils/H2OContextUtils.scala

```diff
@@ -129,62 +122,51 @@ private[spark] trait H2OContextUtils extends Logging {
     } else {
       null
     }
-
-    val outputFileStem = "h2ologs_" + new SimpleDateFormat("yyyyMMdd_hhmmss").format(new Date)
-    zipLogs(perNodeZipByteArray, clientNodeByteArray, outputFileStem, destination)
-  }
-
-  def downloadH2OLogs(destination: String): String = {
-    downloadH2OLogs(new URI(destination))
-    destination
   }
 
-  /** Zip the H2O logs and store them to specified destination */
-  private def zipLogs(results: Array[Array[Byte]], clientResult: Array[Byte], topDir: String, destination: URI): URI = {
-    assert(H2O.CLOUD._memary.length == results.length, "Unexpected change in the cloud!")
-    val l = results.map(_.length).sum
-    val baos = new ByteArrayOutputStream(l)
-    // Add top-level directory.
-    val zos = new ZipOutputStream(baos)
-    val zde = new ZipEntry(topDir + File.separator)
-    zos.putNextEntry(zde)
-
-    try {
-      // Add zip directory from each cloud member.
-      results.zipWithIndex.foreach { case (result, i) =>
-        val filename = topDir + File.separator + "node" + i + "_" + H2O.CLOUD._memary(i).getIpPortString.replace(':', '_').replace('/', '_') + ".zip"
-        val ze = new ZipEntry(filename)
-        zos.putNextEntry(ze)
-        zos.write(result)
-        zos.closeEntry()
-      }
+  /**
+    * @param destination directory where the logs will be downloaded
+    */
+  def downloadH2OLogs(destination: URI, logContainer: LogArchiveContainer): URI = {
+    val workersLogs = getLogsFromWorkers(logContainer)
+    val clientLogs = getLogsFromClient(logContainer)
+    val outputFileStem = "h2ologs_" + new SimpleDateFormat("yyyyMMdd_hhmmss").format(new Date)
 
-      // Add zip directory from the client node.  Name it 'driver' since that's what Sparking Water users see.
-      if (clientResult != null) {
-        val filename = topDir + File.separator + "driver.zip"
-        val ze = new ZipEntry(filename)
-        zos.putNextEntry(ze)
-        zos.write(clientResult)
-        zos.closeEntry()
-      }
-      // Close the top-level directory.
-      zos.closeEntry()
-    } finally {
-      // Close the full zip file.
-      zos.close()
+    val finalArchiveByteArray = try {
+      val method = classOf[RequestServer].getDeclaredMethod("archiveLogs", classOf[LogArchiveContainer],
```

Comment:
It would be nice to change the `archiveLogs` method to public and call the method directly in a different PR.

Reply:
Yup -> https://github.com/h2oai/h2o-3/pull/3698